expression/builtin_cast.go

```diff
@@ -1110,7 +1110,7 @@ func (b *builtinCastStringAsIntSig) handleOverflow(origRes int64, origStr string
 	}
 
 	sc := b.ctx.GetSessionVars().StmtCtx
-	if sc.InSelectStmt && types.ErrOverflow.Equal(origErr) {
+	if types.ErrOverflow.Equal(origErr) {
```

Comment:
It looks like it would only happen in select statements before. Does the change make the insert statement different from mysql?

Reply:
According to the test, this change won't break the existing test. The reason I delete sc.InSelectStmt is that we shouldn't use InSelectStmt to decide how to handle the err. We should use `TruncateAsWarning` , `IgnoreTruncate` to do that.